weatherman/calculations.py

```diff
@@ -0,0 +1,146 @@
+from typing import Dict, Tuple, List
+from collections import defaultdict
+from weatherman.utils import extract_year, extract_year_month
+from weatherman.results import (YearlyExtremeWeatherResult, MonthlyAveragesWeatherResult,
+                                MonthlyExtremeBarChartResult)
+
+
+class WeatherCalculations:
+    """
+    Performs calculations on parsed weather data.
+    """
+    def __init__(self, weather_recordings_data: list[Dict[str, str]]):
+        self.weather_data = weather_recordings_data
+
+    def compute_yearly_extreme_weather(self, year: str) -> YearlyExtremeWeatherResult:
+        weatherdata_filtered_by_year = self.filter_weather_data_by_year(year)
+        max_temp_tuple: Tuple[float, str] = (float('-inf'), '')
+        min_temp_tuple: Tuple[float, str] = (float('inf'), '')
+        max_humid_tuple: Tuple[float, str] = (float('-inf'), '')
+        for weather_entry in weatherdata_filtered_by_year:
+            if weather_entry.get('Max TemperatureC'):
+
+                try:
+                    max_temp_value = float(weather_entry['Max TemperatureC'])
+                    if max_temp_value > max_temp_tuple[0]:
+
+                        max_temp_tuple = (max_temp_value, weather_entry['PKT'])
+                except ValueError:
+                    pass
+
+            if weather_entry.get('Min TemperatureC'):
```

Comment:
```
if weather_entry.get('Min TemperatureC'):

      try:
          min_temp_value = float(weather_entry['Min TemperatureC'])
          if min_temp_value < min_temp_tuple[0]:

              min_temp_tuple = (min_temp_value, weather_entry['PKT'])
      except ValueError:
          pass
```

looks very repetitive, rather do something like following

use switch case and simply make a function pass the name and use it in function